org.kde.optiimage.json

```diff
@@ -0,0 +1,93 @@
+{
+    "id": "org.kde.optiimage",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "6.8",
+    "sdk": "org.kde.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command": "optiimage",
+    "finish-args": [
+        "--filesystem=host",
```

Comment:
why is host needed? it does not use portals?

Reply:
home should be enough

Unfortunately, while we use the portal for selecting files to optimize, when the safe mode is enabled, we write to a new file with a suffix next to the original file. This can't be done with portals :(

Reply:
I initially had used `filesystem=home`, but OptiImage doesn't work for me like that.  The tools it calls (jpegoptim, oxipng, etc.) each give different errors, but they either fail to open existing files or create new files, even though everything I'm doing is in my home folder.  It only started working when I used `filesystem=host`.

Reply:
what errors are you seeing? can you post it?

Reply:
This is what I get if I only have `filesystem=home` and not `filesystem=host`.  Different errors from each tool, but none of them work.

![optiimage-errors-filesystem-home](https://github.com/user-attachments/assets/cb1fe595-851f-49a5-af95-00e20cf58591)

Reply:
is your home `/home/tjk` or `/var/home/tjk`.

on a distro like SilverBlue it's `/var/home` with a symlink for `/home`, which make things unreachable with only home permissions.

Reply:
that doesn't tell much, but if it is under `home` there is no reason `host` should be needed. `host` exposes home + `/run/media /mnt, /opt, /srv`, they don't make a difference here

Reply:
> on a distro like SilverBlue it's `/var/home` with a symlink for `/home`, which make things unreachable with only home permissions.

It's `/var/home`.  So I should change the PR to `filesystem=home` and people with this type of system have to change permissions manually?

Reply:
Yes the permission should be `home`.

No they don't need to change the permission manually. The input path are incorrect. I only ran into the trouble because some path were stored and I downgraded from Workstation (where it is `/home`) to SilverBlue (where it is `/var/home`)

Reply:
> No they don't need to change the permission manually. The input path are incorrect. I only ran into the trouble because some path were stored and I downgraded from Workstation (where it is `/home`) to SilverBlue (where it is `/var/home`)

What should the user do if they have the `/var/home` setup, and this Flatpak with `filesystem=host` gives them the errors I showed?  It seems like adding permissions is the only way to use the tool.  Unless Flatpak has a change coming to better handle that type of filesystem layout.

Reply:
> What should the user do if they have the /var/home 

Then file path in their home directoruy start with `/var/home`

`~` should expand correctly. So do `$HOME`. If not then their distro is misconfigured.

Reply:
why does `host` make any difference? `host` doesn't change or expose anything in `/var`.

It specifically says `/var` is excluded.

```
            host
               The entire host file system, except for directories that
               are handled specially by Flatpak. In particular, this
               shares /home, /media, /opt, /run/media and /srv if they
               exist.

               /dev is not shared: use devices=all; instead.

               Parts of /sys are always shared. This option does not
               make additional files in /sys available.

               Additionally, this keyword provides all of the same
               directories in /run/host as the host-os and host-etc
               keywords. If this keyword is used in conjunction with one
               of the host- keywords, whichever access level is higher
               (more permissive) will be used for the directories in
               /run/host: for example, host:rw;host-os:ro; is equivalent
               to host:rw;.

               These other reserved directories are currently excluded:
               /app, /bin, /boot, /efi, /etc, /lib, /lib32, /lib64,
               /proc, /root, /run, /sbin, /tmp, /usr, /var.

               Available since 0.3.
```

Reply:
> why does host make any difference? host doesn't change or expose anything in /var.

I don't know.  I guess my distro is misconfigured, as hfiguiere said.  For now, I have to add host permissions to run this app.  Whether it's a distro problem or a Flatpak problem, I don't think it's relevant to this application being added, though.

Reply:
> why does `host` make any difference? `host` doesn't change or expose anything in `/var`.

Because. With `home` it export `/var/home`

```
% flatpak run --filesystem=home org.freedesktop.Sdk//24.08
[📦 org.freedesktop.Sdk ~]$ echo $HOME
/var/home/hub
[📦 org.freedesktop.Sdk ~]$ ls -ld /home
ls: cannot access '/home': No such file or directory
[📦 org.freedesktop.Sdk ~]$ ls -ld /var/home/hub/
drwx------. 1 hub hub 7908 Dec 13 21:28 /var/home/hub/
```

With host:

```
[📦 org.freedesktop.Sdk ~]$ ls -ld /home
lrwxrwxrwx. 1 hub hub 8 Dec 13 21:30 /home -> var/home
```

The rest is the same.
That's why it work with `host` but not `home`, because it can resolve the `/home` symlink.

Reply:
Just to clarify. The content of `$HOME` is what matters. The fact the `/home` works by symlink is just compatibility for broken software.

Reply:
sounds like a Flatpak bug.

Reply:
https://github.com/flatpak/flatpak/issues/3582

Reply:
no. it is not.

Reply:
so is this accessing the path like `/home` instead of using `$HOME` or `~/`?

Reply:
that's what the screenshot shows, but I'm not sure how it was initiated.

Reply:
It seems to be using `file://` URIs as Fedora provides the `home` symlinks the URI is `file:///home/foo/bar.png` which is wrong inside the Flatpak.

Reply:
Looks like the difference between home and host, is that `host` shares `/home` (so the symlink gets created as well) while `home` shares `$HOME`.

Reply:
Worked fine here. Selected the file with the file dialog.

`--filesystem=home`.

![image](https://github.com/user-attachments/assets/06c3c3c1-7d07-43eb-90c6-e04584bc2791)

Note that it used the Gtk file dialog, so I suspect it might be using the portal.

And I tested and it works without filesystem permissions.

Reply:
> And I tested and it works without filesystem permissions.

But because of how the portals work the output is stuck in the wrong place.